Gitweb:     http://git.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=3ddf1e7f57237ac7c5d5bfb7058f1ea4f970b661
Commit:     3ddf1e7f57237ac7c5d5bfb7058f1ea4f970b661
Parent:     bb875b38dc5e343bdb696b2eab8233e4d195e208
Author:     Miklos Szeredi <mszeredi@suse.cz>
AuthorDate: Mon Jan 26 15:00:58 2009 +0100
Committer:  Miklos Szeredi <mszeredi@suse.de>
CommitDate: Mon Jan 26 15:00:58 2009 +0100

    fuse: fix missing fput on error
    
    Fix the leaking file reference if allocation or initialization of
    fuse_conn failed.
    
    Signed-off-by: Miklos Szeredi <mszeredi@suse.cz>
    CC: stable@kernel.org

 fs/fuse/inode.c |    9 +++++++--
 1 files changed, 7 insertions(+), 2 deletions(-)

--- a/fs/fuse/inode.c
+++ b/fs/fuse/inode.c
@@ -825,12 +825,16 @@ static int fuse_fill_super(struct super_
 	if (!file)
 		return -EINVAL;
 
-	if (file->f_op != &fuse_dev_operations)
+	if (file->f_op != &fuse_dev_operations) {
+		fput(file);
 		return -EINVAL;
+	}
 
 	fc = new_conn(sb);
-	if (!fc)
+	if (!fc) {
+		fput(file);
 		return -ENOMEM;
+	}
 
 	fc->flags = d.flags;
 	fc->user_id = d.user_id;
